mixnet/topology.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+import random
+from dataclasses import dataclass
+from typing import List, TypeAlias
+
+from cryptography.hazmat.primitives.asymmetric.x25519 import (
+    X25519PrivateKey,
+    X25519PublicKey,
+)
+from pysphinx.node import Node
+
+from mixnet.bls import BlsPrivateKey, BlsPublicKey
+
+
+@dataclass
+class MixnetTopology:
+    # In production, this can be a 1-D array, which is accessible by indexes.
+    # Here, we use a 2-D array for readability.
+    layers: List[List[MixNodeInfo]]
+
+    def generate_route(self) -> list[MixNodeInfo]:
+        return [random.choice(layer) for layer in self.layers]
+
+    def choose_mix_destionation(self) -> MixNodeInfo:
+        all_mixnodes = [mixnode for layer in self.layers for mixnode in layer]
+        return random.choice(all_mixnodes)
+
+
+@dataclass
+class MixnetTopologySize:
```

Comment:
It looks like the "MixnetTopologySize" is used mostly for the robustness layer (and test_utils), so maybe it would be better to make it part of the topology configuration inside of the robustness layer?

Reply:
Resolved by https://github.com/logos-co/nomos-specs/pull/57#discussion_r1469009270